src/charm.py

```diff
@@ -96,18 +94,60 @@ def _are_relations_ready(self) -> bool:
         return self._tls.get_tls_relation() is not None
 
     def _reconcile(self) -> None:
-        """Reconcile charm status based on configuration and integrations."""
+        """Reconcile charm status based on configuration and integrations.
+
+        Raises:
+            RuntimeError: when initializing the lightkube client fails,
+            or when creating the gateway resource fails.
+        """
+        try:
+            # Set field_manager for server-side apply when patching resources
+            # Keep this consistent across client initializations
+            kubeconfig = KubeConfig.from_service_account()
+            client = Client(config=kubeconfig, field_manager=self.app.name)
```

Comment:
You can set the namespace already there and it will be used for any resource action, so you won't need it anymore in your resource controllers

Reply:
PR for this: https://github.com/canonical/gateway-api-integrator-operator/pull/21